keps/3125-maximum-execution-time/README.md

```diff
@@ -0,0 +1,208 @@
+# KEP-3125: Workload Maximum Execution Time
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+  - [Notes/Constraints/Caveats (Optional)](#notesconstraintscaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [API](#api)
+    - [Workload Spec](#workload-spec)
+    - [Workload Status](#workload-status)
+  - [Constants](#constants)
+  - [Controller](#controller)
+    - [Workload](#workload)
+    - [Jobs / Jobframework](#jobs--jobframework)
+  - [Webhooks](#webhooks)
+    - [Workload](#workload-1)
+    - [Jobs](#jobs)
+  - [Test Plan](#test-plan)
+      - [Prerequisite testing updates](#prerequisite-testing-updates)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+      - [controller/core/workload](#controllercoreworkload)
+      - [controller/jobs/job](#controllerjobsjob)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+Add the ability to set a maximum execution time of a Job.
+
+## Motivation
+
+Extend Kueue so that kjobctl can incorporate functionalities equivalent to `-t` in [slurm](https://slurm.schedmd.com/).
+
+Different Jobs CRDs have a field with similar semantics, but there is no standard. For example `batch/Job` has `spec.activeDeadlneSeconds`, while JobSet does not have such an API for now.
+
+### Goals
+
+Introduce a Job agnostic API of setting and enforcing the maximum execution time of a Job.
+
+
+### Non-Goals
+
+- Change the behavior of the Kueue's scheduler based on the maximum execution time.
+- Mitigate potential conflicting configurations between `kueue.x-k8s.io/max-exec-time-seconds` and other job specific `activeDeadline` functionalities.
```

Comment:
I guess that we can expand the `GenericJob` interface or introduce new interface, then we can add the validation to verify if Job does not have `activeDeadlineSeconds` and `kueue.x-k8s.io/max-exec-time-seconds` annotations.

But, users may want to specify both. So, I will postpone this decision to GA criteria. So, could you add an evaluation to to GA criteria if we should extend the JobFramework interface solution to support validations between `kueue.x-k8s.io/max-exec-time-seconds` and other job specific `activeDeadline` functionalities?

Reply:
Oh, sorry. That's true.
I wanted to say the validation implementation in the https://github.com/kubernetes-sigs/kueue/blob/main/pkg/controller/jobs/job/job_webhook.go.

However, you do not plan to implement such validations for now, right?

Reply:
right